lib/pbench/server/api/resources/intake_base.py

```diff
@@ -206,7 +191,7 @@ def _cleanup(self, args: ApiParams, intake: Intake, notes: list[str]):
         pass
 
     def _intake(
-        self, args: ApiParams, request: Request, context: ApiContext
+        self, args: ApiParams, request: Request, _context: ApiContext
```

Comment:
I think that it is kind of ugly and barely cute as documentation.  However, my linter colors the parameters and flags them if they are unused and unprefixed...and I really would like to have the code lint-free (so that, when the linter does flag something, I know not to ignore it).  So, I've been adding the underscores in order to acknowledge when the parameters are expected to be unused.  (Having the linter flag them is, I think, valuable:  outside of mocks, encountering an unused parameter should make one stop and consider if some code is missing or if the interface is properly spec'd; dealing with mocks is a more interesting problem....)

Your point about being able to freely switch between simple argument lists and keyword arguments is well-placed.  However, if the context argument is unused, then there's not much point in the caller specifying it (even to set it to `None`), and the presence of the underscore makes that clear when the caller is being coded...so, I don't think it's really silly.

Reply:
I would dispute that logic: the `_intake` method is at the bottom of a chain descending from `ApiBase._put` (or `_post`) and everyone along the way has access to the context if they need it. The fact that this particular method doesn't currently happen to reference `context` doesn't in any way indicate that a caller shouldn't be providing it.

This also means the names can bounce back and forth (e.g., if we added a new call inside here that *did* require the context), and that's annoying too; especially if we got to the point of generating real API documentation from the docstrings.

I just really dislike the convention, and I see no advantages. It's fine for the IDE to highlight that it isn't used in this context: that's got no bearing on the parameter conventions of the call chain.

Reply:
I certainly concede that the convention (and the linters "enforcement") of it has problems for method which are part of inheritance -- there is no reason to expect that every class's implementation of a given method will necessarily use all of the parameters -- and, obviously, you cannot alter the name of any keyword parameters for methods in the hierarchy.

So, the question is, how do we get the benefits of the linter flagging potential mistakes (or cruft) without interfering with grace of the implementation (and without using comment-based directives, which I think are really ugly).  I don't know.